src/Chainweb/Rosetta/RestAPI.hs

```diff
@@ -85,74 +85,65 @@ data RosettaFailure
     = RosettaChainUnspecified
     | RosettaInvalidChain
     | RosettaMempoolBadTx
-    | RosettaInvalidBlockchainName Text
-    | RosettaMismatchNetworkName ChainwebVersion Text
```

Comment:
If we're not providing dynamic error messages anymore, we should at least keep these two types of bad network errors to improve user debugging.

Reply:
Sure, let's expand `validateNetwork`.